test/unit/test-amp-story-player.js

```diff
@@ -37,6 +37,14 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
 
   const nextTick = () => new Promise((resolve) => win.setTimeout(resolve, 0));
 
+  function afterRenderPromise() {
+    return new Promise((resolve) => {
+      requestAnimationFrame(() => {
+        setTimeout(resolve);
+      });
+    });
+  }
```

Comment:
You're combining 3 different methods to wait for the next frame, or schedule a microtask. Can we find something less hacky?
We have some logic in the implementation to remove the poster and show the image on load, can we use something similar for render? That could be useful to publishers as well.

Reply:
Chatted offline. We will replace these checks when the navigation event is ready.